dashboards-observability/public/components/metrics/sidebar/sidebar.tsx

```diff
@@ -5,73 +5,43 @@
 
 import './sidebar.scss';
 
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { cloneDeep } from 'lodash';
 import { EuiTitle, EuiSpacer, EuiAccordion, EuiLink } from '@elastic/eui';
 import { I18nProvider } from '@osd/i18n/react';
-import { batch, useDispatch } from 'react-redux';
-import { AVAILABLE_METRICS, SELECTED_METRICS } from '../../../../common/constants/metrics';
-import { updateMetrics, sortMetrics, selectMetrics } from '../redux/slices/metrics_slice';
+import { batch, useDispatch, useSelector } from 'react-redux';
+import {
+  availableMetricsSelector,
+  deSelectMetric,
+  selectMetric,
+  loadMetrics,
+  selectedMetricsSelector,
+} from '../redux/slices/metrics_slice';
+import { CoreStart } from '../../../../../../src/core/public';
+import PPLService from '../../../services/requests/ppl';
 
 interface ISidebarProps {
-  metricsList: any;
-  visualizationsList: any;
+  http: CoreStart['http'];
+  pplService: PPLService;
```

Comment:
Explanation for other reviewers :

The more-optimal solution here would be to enclose http and pplService into their own stores (Redux?), so that drill-down is not necessary.   Unfortunately, that work is beyond the resource availability that we have for this release.  For now we'll provide these values in the props here, and the larger refactoring of stores will naturally take care of this case.